src/node/internal/internal_http_server.ts

```diff
@@ -0,0 +1,743 @@
+  414: 'URI Too Long', // RFC 7231 6.5.12
+  415: 'Unsupported Media Type', // RFC 7231 6.5.13
+  416: 'Range Not Satisfiable', // RFC 7233 4.4
+  417: 'Expectation Failed', // RFC 7231 6.5.14
+  418: "I'm a Teapot", // RFC 7168 2.3.3
+  421: 'Misdirected Request', // RFC 7540 9.1.2
+  422: 'Unprocessable Entity', // RFC 4918 11.2
+  423: 'Locked', // RFC 4918 11.3
+  424: 'Failed Dependency', // RFC 4918 11.4
+  425: 'Too Early', // RFC 8470 5.2
+  426: 'Upgrade Required', // RFC 2817 and RFC 7231 6.5.15
+  428: 'Precondition Required', // RFC 6585 3
+  429: 'Too Many Requests', // RFC 6585 4
+  431: 'Request Header Fields Too Large', // RFC 6585 5
+  451: 'Unavailable For Legal Reasons', // RFC 7725 3
+  500: 'Internal Server Error', // RFC 7231 6.6.1
+  501: 'Not Implemented', // RFC 7231 6.6.2
+  502: 'Bad Gateway', // RFC 7231 6.6.3
+  503: 'Service Unavailable', // RFC 7231 6.6.4
+  504: 'Gateway Timeout', // RFC 7231 6.6.5
+  505: 'HTTP Version Not Supported', // RFC 7231 6.6.6
+  506: 'Variant Also Negotiates', // RFC 2295 8.1
+  507: 'Insufficient Storage', // RFC 4918 11.5
+  508: 'Loop Detected', // RFC 5842 7.2
+  509: 'Bandwidth Limit Exceeded',
+  510: 'Not Extended', // RFC 2774 7
+  511: 'Network Authentication Required', // RFC 6585 6
+};
+
+export type DataWrittenEvent = {
+  index: number;
+  entry: WrittenDataBufferEntry;
+};
+
+// @ts-expect-error TS2720 net.Server inconsistencies.
+export class Server
+  extends EventEmitter
+  implements _Server, BaseWithHttpOptions
+{
+  [kConnectionsCheckingInterval]?: number;
+  [kUniqueHeaders]: Set<string> | null = null;
+
+  // Similar option to this. Too lazy to write my own docs.
+  // http://www.squid-cache.org/Doc/config/half_closed_clients/
+  // https://wiki.squid-cache.org/SquidFaq/InnerWorkings#What_is_a_half-closed_filedescriptor.3F
+  httpAllowHalfOpen = false;
+  timeout = 0;
+  maxHeadersCount: number | null = null;
+  maxRequestsPerSocket = 0;
+
+  requestTimeout: number = 0;
+  headersTimeout: number = 0;
+  requireHostHeader: boolean = false;
+  joinDuplicateHeaders: boolean = false;
+  rejectNonStandardBodyWrites: boolean = false;
+  keepAliveTimeout: number = 5_000;
+  port?: number;
+
+  constructor(options?: ServerOptions, requestListener?: RequestListener) {
+    super();
+
+    if (options != null) {
+      storeHTTPOptions.call(this, options);
+    }
+
+    if (typeof options === 'function') {
+      requestListener = options;
+      options = {};
+    } else if (options == null) {
+      options = {};
+    } else {
+      validateObject(options, 'options');
+    }
+
+    if (requestListener) {
+      this.on('request', requestListener);
+    }
+
+    this.on('listening', setupConnectionsTracking);
+
+    this[kUniqueHeaders] = parseUniqueHeadersOption(
+      options.uniqueHeaders as (string | string[])[]
+    );
+  }
+
+  close(): this {
+    httpServerPreClose(this);
+    if (this.port) {
+      portMapper.delete(this.port);
```

Comment:
If this is the only place where `portMapper.delete` is called then we have a possible memory leak issue since `portMapper` is a global map for the worker... if the close method is never actually called then we are leaking. Granted, this is likely not a critical issue as in practice there's likely only going to be one port handler registered per worker but if someone does something odd with this it could become a memory issue for the worker that is difficult to diagnose. Not sure what the fix should be but let's verify that it's not going to become a problem.

Reply:
Yes, I'm not sure what's the best way of doing this. Maybe adding Symbol.dispose to Server, and removing port from portMapper if it exist?

Reply:
I've updated the implementation. Let me know what you think.